bigquery_etl/docs/mozfun/generate_mozfun_docs.py

```diff
@@ -88,7 +88,134 @@ def generate_mozfun_docs(out_dir, project_dir):
                             dataset_doc_file.write(f"{formated}\n\n")
                         # Inject the contents of the README.md
                         dataset_doc_file.write(docfile_content)
-                        # Add links to source and edit
+                        if is_udf:
+                            found_input = False
+                            input_lines = []
+                            found_return = False
+                            return_lines = []
+                            with open(os.path.join(root, UDF_FILE), "r") as udf_sql:
+                                udf_content = [
+                                    line
+                                    for line in udf_sql.readlines()
+                                    if not line.startswith("#")
+                                ]
+
+                                FUNCTION_HEADER = (
+                                    f"CREATE OR REPLACE FUNCTION {dataset_name}.{name}("
+                                )
+                                for line in udf_content:
+                                    # FUNCTION_HEADER starts the UDF and then contains inputs in ()
+                                    if line.startswith(FUNCTION_HEADER) and len(
+                                        line.strip()
+                                    ) > len(FUNCTION_HEADER):
+                                        input_lines.append(line)
+                                    elif line.startswith(FUNCTION_HEADER):
+                                        found_input = True
+                                        input_lines.append(line)
+
+                                    if (
+                                        found_input
+                                        and ") AS (" not in line
+                                        and not line.startswith(FUNCTION_HEADER)
+                                        and "RETURNS" not in line
+                                    ):
+                                        input_lines.append(line)
+                                    elif found_input and (
+                                        ") AS (" in line or "RETURNS" in line
+                                    ):
+                                        found_input = False
+
+                                    # TODO: find_block function? def find_block(start_word, end_word)
+
+                                    # RETURNS has the output types
```

Comment:
We can extend the [function to extract parameters](https://github.com/mozilla/bigquery-etl/blob/3756_function_to_get_UDF_input_output/bigquery_etl/util/udf_functions.py) to get the output parameters using regex and add [test scenarios](https://github.com/mozilla/bigquery-etl/blob/3756_function_to_get_UDF_input_output/tests/util/test_udf_functions.py).